code/modules/library/lib_machines.dm

```diff
@@ -452,25 +452,47 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
 	anchored = 1
 	density = 1
 	var/obj/item/weapon/book/cache		// Last scanned book
+	var/stage = 0

```

Comment:
this is the same variable name as other machine deconstruction procedures, that's why i did it this way